src/libraries/System.Data.Common/ref/System.Data.Common.cs

```diff
@@ -1919,6 +1919,56 @@ public sealed partial class DataTableMappingCollection : System.MarshalByRefObje
         System.Data.ITableMapping System.Data.ITableMappingCollection.Add(string sourceTableName, string dataSetTableName) { throw null; }
         System.Data.ITableMapping System.Data.ITableMappingCollection.GetByDataSetTable(string dataSetTableName) { throw null; }
     }
+    public abstract class DbBatch : System.IDisposable, System.IAsyncDisposable
+    {
+        public System.Data.Common.DbBatchCommandCollection BatchCommands { get { throw null; } }
+        protected abstract System.Data.Common.DbBatchCommandCollection DbBatchCommands { get; }
+        public abstract int Timeout { get; set; }
+        public System.Data.Common.DbConnection? Connection { get; set; }
+        protected abstract System.Data.Common.DbConnection? DbConnection { get; set; }
+        public System.Data.Common.DbTransaction? Transaction { get; set; }
+        protected abstract System.Data.Common.DbTransaction? DbTransaction { get; set; }
+        public System.Data.Common.DbDataReader ExecuteReader(System.Data.CommandBehavior behavior = System.Data.CommandBehavior.Default) { throw null; }
+        protected abstract System.Data.Common.DbDataReader ExecuteDbDataReader(System.Data.CommandBehavior behavior);
+        public System.Threading.Tasks.Task<System.Data.Common.DbDataReader> ExecuteReaderAsync(System.Threading.CancellationToken cancellationToken = default) { throw null; }
+        public System.Threading.Tasks.Task<System.Data.Common.DbDataReader> ExecuteReaderAsync(System.Data.CommandBehavior behavior, System.Threading.CancellationToken cancellationToken = default) { throw null; }
```

Comment:
Why two overloads instead of two defaulted parameters?

(I thought I remembered a discussion about a different new ADO.NET API where it was decided that new code didn't have to follow the existing pattern of multiple overloads. Or maybe I'm remembering the first version of the async schema API and it was eventually changed to follow the pattern and use multiple overloads?)

Reply:
You're right that we could do just one overload with two optional parameters. My logic here was that explicitly specifying the behavior is a relatively rare thing, so it would be good to allow invoking with a cancellation token without having to either specify the behavior or use a named parameter for the token. The same applies for consistency with DbCommand, where you can invoke ExecuteReaderAsync this way.

We did have a discussion about the GetSchemaAsync overloads where we decided to leave them all virtual, but I think that's different (here all the public methods are non-virtual, and call a single abstract protected method without optional parameters).